ethpm_types/contract_type.py

```diff
@@ -204,6 +205,44 @@ def extract_sourcemap_item(expanded_row, item_idx, previous_val=None):
             yield item
 
 
+class PCMapItem(BaseModel):
+    line_start: Optional[int]
+    column_start: Optional[int]
+    line_end: Optional[int]
+    column_end: Optional[int]
+
+
+class PCMap(BaseModel):
+    """
+    As part of the source output, the compiler provides a map of program counter values to
+    statements in the source code that the instructions were compiled from.
+
+    This can be used for various purposes ranging from static analysis tools that
+    report errors based on the program counter value and debugging tools that highlight local
+    variables and their uses.
+    """
+
+    __root__: Dict[str, List[int]]
+
+    def parse(self) -> Dict[int, PCMapItem]:
+        """
+        Parses the pc map string into a map of ``PCMapItem`` items, using integer pc values as keys.
+
+        The format from the compiler will have numeric string keys with lists of ints for values.
+        These integers represent (in order) the starting line, starting column, ending line, and
+        ending column numbers.
+        """
+        return {
+            int(key): PCMapItem.construct(
```

Comment:
`construct()` is for skipping validation, right? What is the reason for skipping validation here? Maybe we should add a comment.

Reply:
Yeah I think I misread Pydantic's docs as "construct builds the object __with__ validation" when it's the opposite 😅 gonna fix this along with the `None` issue mentioned in the other comment

Reply:
Updated in 4a9eb14